cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,567 @@
+        """
+        self._num_cfds_seq = num_cfds_seq
+        self._gnd_state_pops = gnd_state_pops
+
+    @property
+    def data(self) -> Dict[str, Sequence[float]]:
+        return {'number_of_cliffords': self._num_cfds_seq,
+                'ground_state_probability': self._gnd_state_pops}
+
+    def plot(self) -> None:
+        fig = pyplot.figure()
+        pyplot.plot(self._num_cfds_seq, self._gnd_state_pops, 'ro-', figure=fig)
+        pyplot.xlabel(r"Number of Cliffords")
+        pyplot.ylabel('Ground State Probability')
+
+
+class StateTomographyResult(object):
+    """Class for storing and plotting a density matrix obtained from a state
+    tomography experiment."""
+
+    def __init__(self, density_matrix: numpy.ndarray):
+        """
+        Args:
+            density_matrix: The density matrix obtained from tomography.
+        """
+        self._density_matrix = density_matrix
+
+    @property
+    def data(self) -> numpy.ndarray:
+        return self._density_matrix
+
+    def plot(self) -> None:
+        _plot_density_matrix(self._density_matrix)
+
+
+def rabi_oscillations(sampler: sim.SimulatesSamples, qubit: devices.GridQubit,
+                      final_angle: float, num_shots: int, num_points: int) \
+        -> RabiExperimentResult:
+    """
+    Rotates a qubit around the x-axis of the Bloch sphere by a sequence of Rabi
+    angles evenly spaced between 0 and final_angle. For each rotation,
+    repeat the circuit num_shots times and measure the average probability of
+    the qubit being in the |1> state.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubit: The qubit under test.
+        final_angle: The final Rabi angle in units of pi.
+        num_shots: The number of repetitions of the circuit for each Rabi angle.
+        num_points: The number of Rabi angles.
+
+    Returns:
+        A RabiExperimentalResult object that stores and plots the result.
+    """
+    circuit = circuits.Circuit()
+    theta = value.Symbol('theta')
+    circuit.append(ops.X(qubit) ** theta)
+    circuit.append(ops.measure(qubit, key='z'))
+    sweep = study.Linspace(key='theta', start=0.0, stop=final_angle,
+                           length=num_points)
+    results = sampler.run_sweep(circuit, params=sweep, repetitions=num_shots)
+    half_turns = numpy.linspace(0.0, final_angle, num_points)
+    excited_state_pops = numpy.zeros(num_points)
+    for i in range(num_points):
+        excited_state_pops[i] = numpy.mean(results[i].measurements['z'])
+
+    return RabiExperimentResult(num_shots, half_turns, excited_state_pops)
+
+
+def single_qubit_randomized_benchmarking(sampler: sim.SimulatesSamples,
+                                         qubit: devices.GridQubit,
+                                         num_cfds_seq: Sequence[int],
+                                         num_circuits: int, num_shots: int,
+                                         basis='xz') \
```

Comment:
No type annotation on basis. I would prefer if we had IDE-autocompletable arguments such as `use_xy_basis: bool = True`.

Reply:
Done